### PR TITLE
New configuration: .disableOptionDefaultsToNone

### PR DESCRIPTION
**Pull request by jgogstad**
I stuck with the "enabling" semantics in https://github.com/scalalandio/chimney/issues/47.

The code quickly gets confusing with double negations if we go with something like ".disableOptionDefaultsToNone". Like `if (!disableOptionDefaultsToNone)`, but if you're ok with it, I'll make the change.

### CODE REVIEW

File: chimney/src/main/scala/io/scalaland/chimney/internal/TransformerMacros.scala [hunk 1]
```diff
@@ -412,6 +412,14 @@ trait TransformerMacros {
             None
           }
         }
+        .orElse {
+          val targetTypeIsOption = targetField.returnType <:< typeOf[Option[_]]
+          if (targetTypeIsOption && config.optionDefaultsToNone) {
+            Some(ResolvedFieldTree(q"scala.None"))
```
Comment (krzemin) [inline chimney/src/main/scala/io/scalaland/chimney/internal/TransformerMacros.scala]: `_root_.scala.None` would be better

**Comment by krzemin** [comment]
Note that your configuration entry (on the type level or value level) doesn't necessary need to follow semantics of the dsl operator - I'm ok with having them negated so that is dsl you have `.disableOptionDefaultsToNone ` and in the configuration `optionDefaultsToNone`.

**Comment by jgogstad** [comment]
Has the codacy failure been around for a while, or did I introduce that?

**Comment by krzemin** [comment]
Regarding Codacy, we have test coverage of ~89%, but it fails if the PR decreases the previous value. But I see you have already added one more case to prevent that 👍

**Comment by krzemin** [comment]
@Krever I'd like to discuss what defaults are sensible and how we should configure it in a separate issue: https://github.com/scalalandio/chimney/issues/85. For now, just merging as it is.

**Pull request merged**